packages/eslint-plugin/src/util/getBaseTypesOfClassMember.ts

```diff
@@ -0,0 +1,39 @@
+import assert from 'node:assert';
+
+import * as ts from 'typescript';
+
+/**
+ * Given a member of a class which extends another class or implements an interface,
+ * yields the corresponding member type for each of the base class/interfaces.
+ */
+export function* getBaseTypesOfClassMember(
+  checker: ts.TypeChecker,
+  memberTsNode: ts.PropertyDeclaration | ts.MethodDeclaration,
+  heritageToken: ts.SyntaxKind.ExtendsKeyword | ts.SyntaxKind.ImplementsKeyword,
```

Comment:
[Refactor] Both places that call `getBaseTypesOfClassMember` look roughly like:

```ts
for (const { baseMemberType } of util.getBaseTypesOfClassMember(checker, propTsNode, ts.SyntaxKind.ExtendsKeyword) {
    if (isVoidReturningFunctionType(baseMemberType)) {
      reportIfNonVoidFunction(propNode.value);
  }
}

for (const { baseMemberType } of util.getBaseTypesOfClassMember(checker, propTsNode, ts.SyntaxKind.ExtendsKeyword) {
    if (isVoidReturningFunctionType(baseMemberType)) {
      reportIfNonVoidFunction(propNode.value);
  }
}
```

There's some duplication there. I think the `heritageToken` parameter can be removed, to have the function always check both in one call, rather than one each in two calls.

Reply:
Done